CMakeLists.txt

```diff
@@ -0,0 +1,697 @@
+endif()
+
+# Substitute @je_@ in jemalloc_protos.h.in, primarily to make generation of
+# jemalloc_protos_jet.h easy.
+set(je_ je_)
+
+# Do not silence irrelevant compiler warnings
+set(JEMALLOC_CC_SILENCE 1)
+if(disable-cc-silence)
+    set(JEMALLOC_CC_SILENCE 0)
+endif()
+
+# Build debugging code (implies --enable-ivsalloc)
+if(enable-debug)
+    set(JEMALLOC_DEBUG 1)
+    set(JEMALLOC_IVSALLOC 1)
+endif()
+
+# Validate pointers passed through the public API
+if(enable-ivsalloc)
+    set(JEMALLOC_IVSALLOC 1)
+endif()
+
+# Enable stats by default
+set(JEMALLOC_STATS 1)
+# Disable statistics calculation/reporting
+if(disable-stats)
+  set(JEMALLOC_STATS 0)
+endif()
+
+# Enable thread-specific caching by default.
+set(JEMALLOC_TCACHE 1)
+if(disable-tcache)
+  set(JEMALLOC_TCACHE 0)
+endif()
+
+set(JEMALLOC_PREFIX_JET jet_)
+
+# Disabling dss allocation because sbrk is deprecated
+set(JEMALLOC_DSS 0)
+
+# Support the junk/zero filling option by default.
+set (JEMALLOC_FILL 1)
+# Disable support for junk/zero filling, quarantine, and redzones
+if(disable-fill)
+  set (JEMALLOC_FILL 0)
+endif()
+
+# Windows does not have it
+set(JEMALLOC_UTRACE 0)
+set(JEMALLOC_VALGRIND 0)
+
+# Support xmalloc option
+set(JEMALLOC_XMALLOC 0)
+
+if(enable-xmalloc)
+  set(JEMALLOC_XMALLOC 1)
+endif()
+
+# Support cache-oblivious allocation alignment by default.
+# If defined, explicitly attempt to more uniformly distribute large allocation
+# pointer alignments across all cache indices.
+set(JEMALLOC_CACHE_OBLIVIOUS 1)
+
+if(disable-cache-oblivious)
+  set(JEMALLOC_CACHE_OBLIVIOUS 0)
+endif()
+
+# ffsl and ffs are defined in msvc_compat/strings.h
+set(JEMALLOC_INTERNAL_FFSL ffsl)
+set(JEMALLOC_INTERNAL_FFS ffs)
+
+# Base 2 log of minimum tiny size class to support
+set(LG_TINY_MIN 3)
+if(with-lg-tiny-min)
+  set(LG_TINY_MIN ${with-lg-tiny-min})
+endif()
+
+# Base 2 log of minimum allocation alignment
+set(LG_QUANTA 3 4)
+if(with-lg-quantum)
+  # Convert to a CMake list
+  string(REPLACE "," ";" LG_QUANTA  ${with-lg-quantum})
+  set(LG_QUANTA ${with-lg-quantum})
+  set(LG_QUANTUM ${LG_QUANTA})
+endif()
+
+# Base 2 log of system page size
+# Assume default is 4096
+set(SYSTEM_PAGE_SIZE 4096)
+lg(${SYSTEM_PAGE_SIZE} "LG_PAGE")
+if(with-lg-page)
+  set(LG_PAGE ${with-lg-page})
+endif()
+
+## TODO Figure out system page size
+#if(${LG_PAGE} STREQUAL "detect")
+  
+#endif()
+
+# Base 2 logs of system page sizes to support
+set (LG_PAGE_SIZES ${LG_PAGE})
+if(with-lg-page-sizes)
+  string(REPLACE "," ";" LG_PAGE_SIZES  ${with-lg-page-sizes})
+endif()
+
+# Base 2 log of size classes per doubling
+set (LG_SIZE_CLASS_GROUP 2)
+if(with_lg_size_class_group)
+  set (LG_SIZE_CLASS_GROUP ${with_lg_size_class_group})
+endif()
+
+# secure_getenv
+set(JEMALLOC_HAVE_SECURE_GETENV 0)
+
+set(JEMALLOC_HAVE_ISSETUGID 0)
+set(JEMALLOC_MALLOC_THREAD_CLEANUP 0)
+
+set(JEMALLOC_MUTEX_INIT_CB 0)
+
+# No lazy lock since there is a way in Windows
+# to bootstrap for sure
+set(JEMALLOC_LAZY_LOCK 0)
```

Comment:
Now, I am missing je_init in this version which is currently invoked in our in-house version before any static data is initialized.